functions/tokens/[[index]].ts

```diff
@@ -16,11 +17,17 @@ export const onRequest: PagesFunction = async ({ params, request, next }) => {
   const tokenPromise = getToken(networkName, tokenAddress, request.url)
```

Comment:
You shouldn't be getting the token unless you have a cache miss (similarly in other files). Something like:

```
const cachedToken = await getCache(request.url, 'tokens-cache')
if (!cachedToken) {
  const token = await getToken(networkName, tokenAddress, request.url)
  if (!token) return res
  
  putCache(new Response(JSON.stringify(token)), request.url, 'tokens-cache')
  return new HTMLRewriter().on('head', new MetaTagInjector(data)).transform(await res)
}
```